### ISSUE TITLE
Tests?

### ISSUE THREAD

**Issue by garbados** [RangerMauve/pouchdb-adapter-hyperbee#1]
This is a cool plugin for PouchDB but it has [no tests](https://github.com/RangerMauve/pouchdb-adapter-hyperbee/blob/default/package.json#L7)! I hesitate to use it or recommend it as a result.

**Comment by RangerMauve** [comment]
I'd hesitate to use it. 😅 It works okay when you're able to write to the hyperbee, but I've been having issues when trying to read due to the way pouch assumes it can always write.

The project that was funding this has pivoted away from pouch due to the complexity of making it do something it wasn't designed for. 😅 Right now the focus is on [hyperbeedeebee](https://github.com/RangerMauve/hyperbeedeebee) which provides a MongoDB-like interface (similar to pouchdb-find) directly over hyperbee.

**Comment by RangerMauve** [comment]
I could put in a couple of hours to write some basic tests though if people are interested in using it more.

It worked decently up until we needed to use indexes.

**Comment by garbados** [comment]
Ah yes, you'd need some (currently secret) patches to handle dependent DBs on a read-only primary index. Hm.

**Comment by garbados** [comment]
Without revealing too much, there are some pending patches to PouchDB which modify [this line](https://github.com/pouchdb/pouchdb/blob/master/packages/node_modules/pouchdb-core/src/adapter.js#L905) so you can pass custom options to dependent DB construction (indices are dependent DBs), like to use a different storage adapter that can actually be written to. I can't give an estimate on when these patches will land but, they will! Someday!!

**Comment by RangerMauve** [comment]
Yeah, the dependent DBs are exactly where I was running into issues. 😅 If you give me a ping when they're ready I can push some updates to make it work with them.
